src/librustc_typeck/check/intrinsic.rs

```diff
@@ -436,23 +436,29 @@ pub fn check_platform_intrinsic_type(tcx: TyCtxt<'_>, it: &hir::ForeignItem<'_>)
         | "simd_reduce_max"
         | "simd_reduce_min_nanless"
         | "simd_reduce_max_nanless" => (2, vec![param(0)], param(1)),
-        name if name.starts_with("simd_shuffle") => match name["simd_shuffle".len()..].parse() {
-            Ok(n) => {
-                let params = vec![param(0), param(0), tcx.mk_array(tcx.types.u32, n)];
-                (2, params, param(1))
-            }
-            Err(_) => {
-                struct_span_err!(
-                    tcx.sess,
-                    it.span,
-                    E0439,
-                    "invalid `simd_shuffle`, needs length: `{}`",
-                    name
-                )
-                .emit();
-                return;
+        name if {
+            suffix = name.strip_prefix("simd_shuffle");
+            suffix.is_some()
+        } =>
```

Comment:
This works around the fact that if-let guard is not implemented (#51114).
At least it gets rid of repeating the pattern in `starts_with` with may lead to typos.